packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx

```diff
@@ -205,56 +213,79 @@ export class NavBase extends React.Component<INavProps, INavState> implements IN
 
   private _renderGroup = (group: INavLinkGroup, groupIndex: number): React.ReactElement<{}> => {
     const { styles, groups, theme, onRenderGroupHeader = this._renderGroupHeader } = this.props;
+
+    const isExpanded = this._isGroupExpanded(group);
+
     const classNames = getClassNames(styles!, {
       theme: theme!,
       isGroup: true,
-      isExpanded: this._isGroupExpanded(group),
+      isExpanded,
       groups,
     });
 
+    const finalOnHeaderClick = (
+      ev?: React.MouseEvent<HTMLElement, MouseEvent> | undefined,
+      isCollapsing?: boolean | undefined,
+    ) => {
+      this._onGroupHeaderClicked(group, ev);
+    };
+
+    const groupProps: IRenderGroupHeaderProps = {
```

Comment:
As I recently learned with https://github.com/microsoft/fluentui/pull/12750 doing this can cause unwanted rerenders as each group is no longer considered the same object. OTOH we don't want to mutate state either. @xugao do you have any ideas?

Reply:
in our default render for group header, we don't rely on any memoized props. so in that sense, this code is fine. But let's say user has a very expensive customized group header, the header IS receiving an always-mutating props. However i don't see how we can avoid the mutation here